src/libraries/System.Console/src/System/ConsolePal.TerminalFormatStrings.cs

```diff
@@ -0,0 +1,255 @@
+            AddKey(db, TermInfo.WellKnownStrings.KeySelect, ConsoleKey.Select);
+            AddKey(db, TermInfo.WellKnownStrings.KeySLeft, ConsoleKey.LeftArrow, shift: true, alt: false, control: false);
+            AddKey(db, TermInfo.WellKnownStrings.KeySPrint, ConsoleKey.Print, shift: true, alt: false, control: false);
+            AddKey(db, TermInfo.WellKnownStrings.KeySRight, ConsoleKey.RightArrow, shift: true, alt: false, control: false);
+            AddKey(db, TermInfo.WellKnownStrings.KeyUp, ConsoleKey.UpArrow);
+            AddPrefixKey(db, "kLFT", ConsoleKey.LeftArrow);
+            AddPrefixKey(db, "kRIT", ConsoleKey.RightArrow);
+            AddPrefixKey(db, "kUP", ConsoleKey.UpArrow);
+            AddPrefixKey(db, "kDN", ConsoleKey.DownArrow);
+            AddPrefixKey(db, "kDC", ConsoleKey.Delete);
+            AddPrefixKey(db, "kEND", ConsoleKey.End);
+            AddPrefixKey(db, "kHOM", ConsoleKey.Home);
+            AddPrefixKey(db, "kNXT", ConsoleKey.PageDown);
+            AddPrefixKey(db, "kPRV", ConsoleKey.PageUp);
+
+            if (KeyFormatToConsoleKey.Count > 0)
+            {
+                MaxKeyFormatLength = int.MinValue;
+                MinKeyFormatLength = int.MaxValue;
+
+                foreach (KeyValuePair<ReadOnlyMemory<char>, ConsoleKeyInfo> entry in KeyFormatToConsoleKey)
+                {
+                    if (entry.Key.Length > MaxKeyFormatLength)
+                    {
+                        MaxKeyFormatLength = entry.Key.Length;
+                    }
+                    if (entry.Key.Length < MinKeyFormatLength)
+                    {
+                        MinKeyFormatLength = entry.Key.Length;
+                    }
+                }
+            }
+        }
+
+        private static string GetTitle(TermInfo.Database db)
+        {
+            // Try to get the format string from tsl/fsl and use it if they're available
+            string? tsl = db.GetString(TermInfo.WellKnownStrings.ToStatusLine);
+            string? fsl = db.GetString(TermInfo.WellKnownStrings.FromStatusLine);
+            if (tsl != null && fsl != null)
+            {
+                return tsl + "%p1%s" + fsl;
+            }
+
+            string term = db.Term;
+            if (term == null)
+            {
+                return string.Empty;
+            }
+
+            if (term.StartsWith("xterm", StringComparison.Ordinal)) // normalize all xterms to enable easier matching
+            {
+                term = "xterm";
+            }
+
+            switch (term)
+            {
+                case "aixterm":
+                case "dtterm":
+                case "linux":
+                case "rxvt":
+                case "xterm":
+                    return "\x1B]0;%p1%s\x07";
+                case "cygwin":
+                    return "\x1B];%p1%s\x07";
+                case "konsole":
+                    return "\x1B]30;%p1%s\x07";
+                case "screen":
```

Comment:
Should we look at variants of `screen`, specially `screen-256color` (used by [tmux](https://github.com/tmux/tmux/wiki/FAQ)) ?

Reply:
@omajid Great catch! I've added the "normalization" and recorded test data for such tmux config